pkgs/top-level/stage.nix

```diff
@@ -211,6 +211,20 @@ let
       };
     };
 
+    pkgsBolt = nixpkgsFun {
```

Comment:
Do we really need this? I'm not sure every possible thing you can set on a platform needs its own package set, and I especially don't think we need Hydra to build every single-change stdenv.

Reply:
We can drop it before merging but for the purposes of testing this PR, I added it in.

Reply:
I think this sort of thing is useful to have. For what it's worth. I don't think its existence needs to promise that it will work, much the same as for pkgsStatic and friends as well. It is quite powerful to have a simple name for this thing to reach for from the shell.

Reply:
The problem is that these things also make it way too easy to use them _within_ Nixpkgs, and then we end up with a layering violation where a single package is going through a whole distinct package set out of convenience.

But that's a general problem that needs to be solved, by making these inaccessible from _within_ Nixpkgs while still being a convenience layer on top, so if this is really useful it can stay.  I think there's no justification for building this set on Hydra, though.

Reply:
> The problem is that these things also make it way too easy to use them within Nixpkgs, and then we end up with a layering violation where a single package is going through a whole distinct package set out of convenience.

We can add in a config option similar to the aliases one which enables the subsets. Then have it disabled in Hydra / CI.

> But that's a general problem that needs to be solved, by making these inaccessible from within Nixpkgs while still being a convenience layer on top, so if this is really useful it can stay. I think there's no justification for building this set on Hydra, though.

I kinda agree but its also good to ensure things work.

Reply:
That relies on somebody actively monitoring that on Hydra indefinitely, which has not historically happened for this sort of thing.  Hydra isn't really a great system for "ensuring things work" due to the lack of notifications.

Reply:
> > The problem is that these things also make it way too easy to use them within Nixpkgs, and then we end up with a layering violation where a single package is going through a whole distinct package set out of convenience.
> 
> We can add in a config option similar to the aliases one which enables the subsets. Then have it disabled in Hydra / CI.

Oh that's not a bad idea!

Reply:
This is done in #400351

Reply:
I thought it was resolved because it sounded like Alyssa agreed to the variants overlay.

Reply:
I still don't really like adding new ones, but not enough to really object if it can't be used within Nixpkgs.